domain/state.go

```diff
@@ -15,39 +15,50 @@ import (
 // StateBase defines a base struct for requesting a database. This will cache
 // the database for the lifetime of the struct.
 type StateBase struct {
-	mu    sync.Mutex
-	getDB database.TxnRunnerFactory
-	db    database.TxnRunner
+	dbMutex sync.RWMutex
```

Comment:
Could have left this out of the churn.

Reply:
I had a conversation about this the other day. I asked if I should add it, you said yes, and here it is.